src/streamer.rs

```diff
@@ -489,25 +522,15 @@ fn recv_window(
             let p = b.write().expect("'b' write lock in fn recv_window");
             (p.get_index()?, p.meta.size)
         };
-        // Prevent receive window from running over
-        if pix >= *consumed + WINDOW_SIZE {
-            recycler.recycle(b);
-            continue;
-        }
-        if pix > *received {
-            *received = pix;
-        }
-        // Got a blob which has already been consumed, skip it
-        // probably from a repair window request
-        if pix < *consumed {
-            debug!(
-                "{:x}: received: {} but older than consumed: {} skipping..",
-                debug_id, pix, *consumed
-            );
+
+        let result = validate_blob_against_window(debug_id, pix, *consumed, *received);
+        if result.is_none() {
```

Comment:
Can you use a `match` here? Under the hood, both `is_none()` and `result.unwrap()` below would use a match. I'm guessing clippy is about to suggest this as well. Let's see!

Reply:
Funny, as I used match initially, and then changed it to this. I'll change it to match.